### PR TITLE
parser: disallow 'SET tracing' in some contexts

### LINKED ISSUE #81410: sql: `ALTER ROLE ... SET tracing=on;` crashes the database
**Describe the problem**
```
CREATE USER roacher;
ALTER ROLE roacher SET tracing = 'on';
```

crashes the database with message:
```
invalid syntax: statement ignored: unexpected error: driver: bad connection
warning: connection lost!
opening new connection: all session settings will be lost
warning: error retrieving the transaction status: dial tcp [::1]:26257: connect: connection refused
warning: connection lost!
opening new connection: all session settings will be lost
warning: error retrieving the database name: dial tcp [::1]:26257: connect: connection refused
root@localhost:26257/? ?>
```

### PR DESCRIPTION
**Pull request by rafiss**
fixes https://github.com/cockroachdb/cockroach/issues/81410

Release note (bug fix): Fixed a panic that was caused by setting the
"tracing" session variable using SET LOCAL or ALTER ROLE ... SET.

### CODE REVIEW

File: pkg/sql/sem/tree/set.go [hunk 1]
```diff
@@ -128,6 +151,11 @@ type SetTracing struct {
 	Values Exprs
 }
 
+// GetSetVar implements the SetClause interface.
+func (node *SetTracing) GetSetVar() (*SetVar, error) {
+	return nil, pgerror.Newf(pgcode.Syntax, "cannot set \"tracing\"  in this context")
```
Comment (otan) [inline pkg/sql/sem/tree/set.go:156]: nit: extra space, you can avoid the escape by using backticks:

```
	return nil, pgerror.Newf(pgcode.Syntax, `cannot set "tracing" in this context`)

```

(also wonder whether

```

	return pgerror.Newf(pgcode.CantChangeRuntimeParam,
		"parameter %q cannot be changed", varName)
```

is better)

**Review by otan — APPROVED** [review]
i'm fine with this; but as added security want to add `tracing` to https://github.com/cockroachdb/cockroach/blob/d5170bbb9feca28513c0ce7ca36d8d174e84370e/pkg/sql/alter_role.go#L377 anyway? :)

**Comment by rafiss** [comment]
@otan i made this less hacky now. could you review again plz?

**Review by otan — APPROVED** [review]
this is nice, i like

**Comment by rafiss** [comment]
tftr!

bors r=otan

**Comment by rafiss** [comment]
had to fix a grammar rule to make `SHOW tracing` work correctly.

bors r=otan

**Pull request merged**